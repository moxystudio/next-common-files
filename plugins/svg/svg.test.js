const { withSVG } = require('../../index.js');

const options = {
    dev: true,
    isServer: true,
};

const mockFunction = jest.fn();

let config;

beforeEach(() => {
    config = {
        module: {
            rules: [],
        },
    };

    jest.clearAllMocks();
});

it('should match snapshot for all results', () => {
    const arrayOfSVGRulesWithInline = [];
    const arrayOfSVGRulesWithDefault = [];

    arrayOfSVGRulesWithInline.push(withSVG({ inline: true })().webpack(config, options).module.rules);
    arrayOfSVGRulesWithDefault.push(withSVG()().webpack(config, options).module.rules);

    expect(arrayOfSVGRulesWithInline).toMatchSnapshot();
    expect(arrayOfSVGRulesWithDefault).toMatchSnapshot();
});

it('should return a single rule', () => {
    const arrayOfSVGRules = [];

    arrayOfSVGRules.push(withSVG()().webpack(config, options).module.rules);

    expect(arrayOfSVGRules).toHaveLength(1);
});

it('should return a function if next config is a function', () => {
    const nextConfig = {
        webpack: mockFunction,
    };

    withSVG()(nextConfig).webpack(config, options);

    expect(nextConfig.webpack.mock.calls).toHaveLength(1);
});

it('should change loaders in SVG in accordance with inline option', () => {
    const arrayOfSVGRulesWithInline = [];
    const arrayOfSVGRulesWithDefault = [];

    arrayOfSVGRulesWithInline.push(withSVG({ inline: true })().webpack(config, options).module.rules);
    arrayOfSVGRulesWithDefault.push(withSVG()().webpack(config, options).module.rules);

    expect(arrayOfSVGRulesWithInline).not.toBe(arrayOfSVGRulesWithDefault);
});

it('should have inline: false as default', () => {
    const arrayOfSVGRulesWithFalse = [];
    const arrayOfSVGRulesWithDefault = [];

    arrayOfSVGRulesWithFalse.push(withSVG({ inline: false })().webpack(config, options).module.rules);
    arrayOfSVGRulesWithDefault.push(withSVG()().webpack(config, options).module.rules);

    expect(arrayOfSVGRulesWithFalse).toEqual(arrayOfSVGRulesWithDefault);
});
