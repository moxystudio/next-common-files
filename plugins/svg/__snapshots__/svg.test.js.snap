// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot for all results 1`] = `
Array [
  Array [
    Object {
      "test": /\\\\\\.svg\\$/,
      "use": Array [
        Object {
          "loader": "raw-loader",
        },
        Object {
          "loader": "svgo-loader",
          "options": Object {
            "plugins": Array [
              Object {
                "removeTitle": true,
              },
              Object {
                "removeDimensions": false,
              },
              Object {
                "removeViewBox": false,
              },
              Object {
                "cleanupIDs": false,
              },
            ],
          },
        },
        Object {
          "loader": "svg-css-modules-loader",
          "options": Object {
            "transformId": true,
          },
        },
      ],
    },
    Object {
      "loader": "url-loader",
      "options": Object {
        "emitFile": false,
        "limit": 0,
        "name": "images/[name].[ext]",
        "outputPath": "static/chunks/media",
        "publicPath": "undefined_next/static/chunks/media",
      },
      "test": /\\\\\\.svg\\$/,
    },
  ],
]
`;

exports[`should match snapshot for all results 2`] = `
Array [
  Array [
    Object {
      "test": /\\\\\\.svg\\$/,
      "use": Array [
        Object {
          "loader": "raw-loader",
        },
        Object {
          "loader": "svgo-loader",
          "options": Object {
            "plugins": Array [
              Object {
                "removeTitle": true,
              },
              Object {
                "removeDimensions": false,
              },
              Object {
                "removeViewBox": false,
              },
              Object {
                "cleanupIDs": false,
              },
            ],
          },
        },
        Object {
          "loader": "svg-css-modules-loader",
          "options": Object {
            "transformId": true,
          },
        },
      ],
    },
    Object {
      "loader": "url-loader",
      "options": Object {
        "emitFile": false,
        "limit": 0,
        "name": "images/[name].[ext]",
        "outputPath": "static/chunks/media",
        "publicPath": "undefined_next/static/chunks/media",
      },
      "test": /\\\\\\.svg\\$/,
    },
  ],
]
`;
